providers/monitor.rb

```diff
@@ -72,3 +70,13 @@ def whyrun_supported?
     provider service_provider unless service_provider.nil?
   end
 end
+
+private
+
+def yaml_dir
+  if node['datadog']['agent6']
+    ::File.join(node['datadog']['agent6_config_dir'], 'conf.d', "#{new_resource.name}.d")
```

Comment:
could you remove the `"#{new_resource.name}.d"` part to keep only `::File.join(node['datadog']['agent6_config_dir'], 'conf.d')`? Agent 6 still supports that configuration scheme (and that's the scheme the cookbook currently uses for Linux Agent 6). Changing the scheme now would slightly break the cookbook for Linux Agent 6 users upgrading from previous versions of the cookbook.

I'm planning to switch to the new scheme in the next major version of the cookbook.